pkg/operator/petset.go

```diff
@@ -157,8 +165,10 @@ func makePetSetSpec(name, image, version, retention string, replicas int32, aler
 							},
 						},
 						Args: []string{
-							"-storage.local.retention=" + retention,
-							"-storage.local.memory-chunks=500000",
+							"-storage.local.retention=" + p.Spec.Retention,
+							"-storage.local.memory-chunks=" + fmt.Sprintf("%d", memChunks),
+							"-storage.local.max-chunks-to-persist" + fmt.Sprintf("%d", memChunks/2),
+							"-storage.local.num-fingerprint-mutexes=4096",
```

Comment:
Ah, I didn't even check the default. Unnecessary but being explicit about it doesn't hurt I suppose.